src/alchemlyb/tests/parsing/test_gmx.py

```diff
@@ -180,8 +180,9 @@ def _diag_sum(dataset):
             u_nk = extract_u_nk(filename, T=300)
 
             # Calculate the sum of diagonal elements:
-            for i in range(len(dataset['data'][leg])):
-                ds += u_nk.iloc[i][i]
+            for i, lambda_ in enumerate(u_nk.columns):
+                #18.6 is the time step
+                ds += u_nk.loc[i*186/10][lambda_].values[0]
```

Comment:
If this is a proper `.loc`, can't we use
```suggestion
                ds += u_nk.loc[i*186/10, lambda_].values[0]
```

Reply:
This is not a proper `.loc`. 
the .loc[1.0] could in theory match both dataframe with row index (1.0, 0.0, 0.0) and (1.0).
But .loc[1.0, :] would only strictly match (1.0).

Reply:
Ok, after looking at the data I see that this is a multiindex. But in this case wouldn't be
```
ds += u_nk.loc[i*186/10].loc[lambda_].values[0]
```
be cleaner?

(I have no idea how to get something like ".loc[lambda_, (0.0, 0.0)]" instead of the values[0] but I find pandas indexing confusing...)

Reply:
Yes, but we cannot get lambda_ beforehand and the lambda_ could be (0.0, 0.0) or (0.0, 0.0, 0.0), which is why for some of them, I could only do `iloc`.
In some sense, we could get them by analysing the dataframe but that would make the unit test too complicated so I think the current form is a good balance.